doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -0,0 +1,81 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+/**
+ * @defgroup auto_symb_diff Automatic and symbolic differentiation
```

Comment:
@davydden Here's an initial outline of a module based on my description in the PR. Although I'll be adding to this as I go along I hope that you find it to be a decent starting point.

Reply:
great, i will have a look!